libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/view/MapboxManeuverView.kt

```diff
@@ -75,7 +80,7 @@ class MapboxManeuverView : ConstraintLayout {
     private val mainLayoutBinding = MapboxMainManeuverLayoutBinding.bind(binding.root)
     private val subLayoutBinding = MapboxSubManeuverLayoutBinding.bind(binding.root)
 
-    private val maneuverToRoadShields = mutableMapOf<String, List<RoadShield>>()
+    private val maneuverToRoadShields = mutableSetOf<RoadShield>()
```

Comment:
This should rather be `mutableSetOf<RouteShieldResult>()` and we should map from the old `RoadShield` to `RouteShieldResult` for the deprecated APIs.

With that, we should deprecate the `RoadShield` itself since all of the APIs that rely on it are deprecated as well.